.github/workflows/build.yaml

```diff
@@ -0,0 +1,33 @@
+name: Build
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: ${{ matrix.os }} • ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ["3.11", "3.12", "3.13"]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Setup uv and cache packages
+        uses: astral-sh/setup-uv@v5
+      - run: uv build
+      - name: Set up isolated venv and test import
+        shell: bash
+        run: |
+          uv venv /tmp/testenv
+          if [[ "$RUNNER_OS" == "Windows" ]]; then
+            . /tmp/testenv/Scripts/activate
+          else
+            source /tmp/testenv/bin/activate
+          fi
+          uv pip install dist/*.whl
+          python -c "import browser_use; print('browser_use package is imported and functional!')" 
```

Comment:
so the `build` should actually happen first without any matrix, just setup python 3.12 and uv and build, and push the build output as an artifact. in theory building the package should never change across python versions since it's not actually interpreting any library code during build, it's just bundling according to uv/hatch logic.

Then the `build_test` test should re-use the previously built artifact and import it across the whole OS/python-version matrix + test `python -c 'from browser_use import Agent, Browser, Controller, ActionModel, ActionResult'` on each os/version combo.